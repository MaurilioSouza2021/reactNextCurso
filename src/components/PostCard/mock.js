export const postCardPropsMock = {
  title: 'Title1',
  body: 'body 1',
  id: 1,
  cover: 'img/img.png',
};
