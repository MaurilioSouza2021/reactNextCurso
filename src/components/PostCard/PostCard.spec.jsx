import { render, screen } from '@testing-library/react';
import { PostCard } from '.';
import { postCardPropsMock as props } from './mock';

describe('<PostCard />', () => {
  it('should render PostCard correctly', () => {
    render(<PostCard {...props} />);
    expect(screen.getByRole('img', { name: props.title })).toHaveAttribute('src', props.cover);
    expect(screen.getByRole('heading', { name: `${props.id} ${props.title}` })).toBeInTheDocument();
    expect(screen.getByText('body 1')).toBeInTheDocument();
  });

  it('should match snapshot', () => {
    const { container } = render(<PostCard {...props} />);
    expect(container.childNodes).toMatchSnapshot();
  });
});
