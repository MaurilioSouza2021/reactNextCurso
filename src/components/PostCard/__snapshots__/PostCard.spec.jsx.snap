// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostCard /> should match snapshot 1`] = `
NodeList [
  <div
    class="post"
  >
    <img
      alt="Title1"
      src="img/img.png"
    />
    <div
      class="post-content"
    >
      <h2>
        1
         
        Title1
      </h2>
      <p>
        body 1
      </p>
    </div>
  </div>,
]
`;
